components/neeva/renderer/content_filtering_agent.cc

```diff
@@ -149,6 +149,13 @@ ContentFilteringPolicy ContentFilteringAgent::GetPolicyForRequest(
 
     // A match was found!
     *rules_name = filter.rules_name;
+    // There's no point in blocking foo.com from making requests from first-party origins (e.g. foo.com). 
+    // Doing so can break logins or other site functionality. 
+    if (!IsThirdParty(url, first_party_origin)) {
```

Comment:
nit: The computation of `IsThirdParty` already just happened. It would be reasonable to save that in a variable above so we don't have to repeat it here.